scene/gui/color_picker.cpp

```diff
@@ -295,7 +295,12 @@ void ColorPicker::_update_controls() {
 }
 
 void ColorPicker::_set_pick_color(const Color &p_color, bool p_update_sliders) {
+	if (color == p_color) {
+		return;
+	}
 	color = p_color;
+	emit_signal(SNAME("color_changed"), color);
```

Comment:
The signal shouldn't be emitted here, because when it makes it emitted when changing color via code.